internal/cli/cmd/root.go

```diff
@@ -21,19 +21,53 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/suse/elemental/v3/pkg/log"
+	"github.com/suse/elemental/v3/pkg/sys"
 	"github.com/urfave/cli/v2"
 )
 
+type GlobalFlags struct {
+	Debug bool
+}
+
+var GlobalArgs GlobalFlags
+
 func appName() string {
 	return filepath.Base(os.Args[0])
 }
 
+func globalFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.BoolFlag{
+			Name:        "debug",
+			Usage:       "Set logging at debug level",
+			Destination: &GlobalArgs.Debug,
+		},
+	}
+}
+
 func NewApp() *cli.App {
 	app := cli.NewApp()
 
+	app.Flags = globalFlags()
 	app.Name = appName()
 	app.Usage = "Build, install and upgrade infrastructure platforms"
 	app.Suggest = true
+	app.Before = func(ctx *cli.Context) error {
+		s, err := sys.NewSystem()
+		if err != nil {
+			return err
+		}
+
+		if ctx.Bool("debug") {
+			s.Logger().SetLevel(log.DebugLevel())
+		}
+		if ctx.App.Metadata == nil {
+			ctx.App.Metadata = map[string]any{}
+		}
+		ctx.App.Metadata["system"] = s
```

Comment:
```suggestion
		ctx.Context = context.WithValue(ctx.Context, "system", s)
```

I'm wondering if the context itself makes a better use of passing this value around. It is more common and I can see both working so I'm just leaving the idea if you like it.

Reply:
Since this is just used to workaround urfave I'd prefer to stick with the urfave utilites for that matter (this is just passing data from root command to subcommands). The Context inside the urfave context is likely to go down deeper in the code and there is not actually needed a context with values. In former `pkg/sys` PR it was already discussed as a preferred approach passing data as explicit arguments.
So to ensure the System is not pulled from the context down the road I think it is just better to not include it there. The `App.Metadata` will only be used in the `internal/cmd` only.